src/rpm/package.rs

```diff
@@ -849,15 +867,20 @@ impl PackageMetadata {
                     mtimes,
                     sizes,
                     flags,
+                    caps,
                 ))
                 .try_fold::<Vec<FileEntry>, _, Result<_, Error>>(
                     Vec::with_capacity(n),
-                    |mut acc, (path, user, group, mode, digest, mtime, size, flags)| {
+                    |mut acc, (path, user, group, mode, digest, mtime, size, flags, _cap)| {
                         let digest = if digest.is_empty() {
                             None
                         } else {
                             Some(FileDigest::load_from_str(algorithm, digest)?)
                         };
+                        let cap = match caps {
+                            Some(ref caps) => caps.get(acc.len()).map(|x| x.to_owned()),
+                            None => None
+                        };
```

Comment:
I suppose this works, but do you think indexing directly would be cleaner? i.e. drop changes on line 870 and 874, use enumerate on the iterator to get a direct index, and fetch the index from `caps`

I think it would be a little more complicated than that but that's the basic idea.  The reason I suggest it is that there's probably a few other bits of metadata that would work this way, and that approach feels potentially easier to scale than this one.